libraries/Microsoft.Bot.Builder/Skills/SkillHandlerImpl.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Claims;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Bot.Builder.Skills
+{
+    internal class SkillHandlerImpl
+    {
+        private readonly string _skillConversationReferenceKey;
+        private readonly BotAdapter _adapter;
+        private readonly IBot _bot;
+        private readonly SkillConversationIdFactoryBase _conversationIdFactory;
+        private readonly Func<string> _getOAuthScope;
+        private readonly ILogger _logger;
+
+        internal SkillHandlerImpl(
+            string skillConversationReferenceKey,
+            BotAdapter adapter,
+            IBot bot,
+            SkillConversationIdFactoryBase conversationIdFactory,
+            Func<string> getOAuthScope,
+            ILogger logger = null)
+        {
+            _skillConversationReferenceKey = skillConversationReferenceKey ?? throw new ArgumentNullException(nameof(skillConversationReferenceKey));
```

Comment:
this is a string parameter, validate against NullOrEmpty to make sure a valid key is passed.